packages/client/utils/getTokenFromSSO.ts

```diff
@@ -1,30 +1,52 @@
 import getOAuthPopupFeatures from './getOAuthPopupFeatures'
+import {GA4SignUpEventEmissionRequiredArgs} from './handleSuccessfulLogin'
 
-const getTokenFromSSO = (url: string) => {
+type SucessReturnType = {
+  token: string
+  ga4Args: GA4SignUpEventEmissionRequiredArgs
+}
+
+type ErrorReturnType = {
+  error: string
+}
+
+type ReturnType = SucessReturnType | ErrorReturnType
```

Comment:
Let me know if this is a better way to handle your last 2 comments, or it's overkill

Reply:
Yes, looks good!